File: AmaroTest/app/src/main/java/com/test/amaro/amarotest/dagger/components/ActivityComponent.java
package com.test.amaro.amarotest.dagger.components;

import com.test.amaro.amarotest.dagger.modules.PresenterModule;
import com.test.amaro.amarotest.dagger.scopes.PerActivity;
import com.test.amaro.amarotest.presentation.detail.DetailActivity;
import com.test.amaro.amarotest.presentation.main.MainActivity;
import dagger.Subcomponent;

@PerActivity
@Subcomponent(modules = PresenterModule.class)
public interface ActivityComponent {

    void inject(MainActivity mainActivity);
    void inject(DetailActivity detailActivity);
}

File: AmaroTest/app/src/main/java/com/test/amaro/amarotest/dagger/components/ActivityComponent.kt
package com.test.amaro.amarotest.dagger.components

import com.test.amaro.amarotest.dagger.modules.PresenterModule
import com.test.amaro.amarotest.dagger.scopes.PerActivity
import com.test.amaro.amarotest.main.MainActivity
import dagger.Subcomponent


/**
 * This component inject dependencies to all Activities across the application
 */
@PerActivity
@Subcomponent(modules = arrayOf(PresenterModule::class))
interface ActivityComponent {

  fun inject(mainActivity: MainActivity)

}